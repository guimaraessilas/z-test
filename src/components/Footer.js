import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { numberToBLR } from "../utils/helper";
import { useSelector } from "react-redux";

const Footer = () => {
  const amount = useSelector((state) => state);
  return (
    <View style={styles.footer}>
      <Text style={styles.footerText}>Total: {numberToBLR(amount)}</Text>
    </View>
  );
};

export default Footer;

const styles = StyleSheet.create({
  footer: {
    height: 70,
    width: "100%",
    backgroundColor: "#0088cc",
    justifyContent: "center",
    alignItems: "center",
  },
  footerText: {
    fontSize: 26,
    textAlign: "center",
    color: "white",
    fontWeight: "700",
  },
});
