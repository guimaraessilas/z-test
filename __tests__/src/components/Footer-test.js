import React from "react";
import Footer from "../../../src/components/Footer";
import renderer from "react-test-renderer";
import { useSelector } from "react-redux";

jest.mock("react-redux");

describe("Should test Footer Component ", () => {
  beforeAll(() => {
    useSelector.mockImplementation(() => 0);
  });

  describe("snapshot", () => {
    it("render correctly", () => {
      const tree = renderer.create(<Footer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
