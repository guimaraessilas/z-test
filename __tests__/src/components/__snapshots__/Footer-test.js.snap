// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test Footer Component  snapshot render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0088cc",
      "height": 70,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 26,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
  >
    Total: 
    R$ 0,00
  </Text>
</View>
`;
